Automattic/wp-calypso
# Odyssey Stats: wire up purchase page

## Proposed Changes

The jump between WP Admin to Calypso blue is a well documented area of confusion for users. We get this comment all the time. So we are going to enable the page within Jetpack.

The PR only wire the page up, and doesn't handle anything else, so you'd expect some of the things not working, which would be fine at the stage.

- Adds an Odyssey implementation of `QueryProductsList`, which queries `jetpack/v4/products`
- Fixed a critical error for Odyssey implementation of `QuerySitePurchases`
- Wired up stats page in Odyssey Stats

Idea for future improvements

- We could add a mechanism to list the API and the origin of the requests, so that we don't have to override the components on build anymore

## Testing Instructions

* Build Jetpack if needed `jetpack build plugins/jetpack`
* Build Odyssey Stats `STATS_PACKAGE_PATH=~/path/to/jetpack/projects/packages/stats-admin yarn dev`
* Open `/wp-admin/admin.php?page=stats#!/stats/purchase/:siteSlug`
* Ensure you see the purchase page

![image](https://github.com/Automattic/wp-calypso/assets/1425433/d3311520-e737-4d5f-bd0c-b178dce3b841)

**Note: the buttons to purchase Stats do not work at the moment and we'll fix those in follow up PRs.**


## Pre-merge Checklist
